tests/schema/__snapshots__/atomicOperations.test.ts.snap

```diff
@@ -0,0 +1,373 @@
+import * as Prisma from '@prisma/client';
+
+// Pagination type
+type Pagination = {
+  first?: boolean
+  last?: boolean
+  before?: boolean
+  after?: boolean
+}
+
+// Prisma custom scalar names
+type CustomScalars = 'No custom scalars are used in your Prisma Schema.'
+
+// Prisma model type definitions
+interface PrismaModels {
+  ModelName: Prisma.ModelName
+}
+
+// Prisma input types metadata
+interface NexusPrismaInputs {
+  Query: {
+    modelNames: {
+      filtering: 'AND' | 'OR' | 'NOT' | 'id' | 'name' | 'amount'
+      ordering: 'id' | 'name' | 'amount'
+    }
+  },
+  ModelName: {
+
+  }
+}
+
+// Prisma output types metadata
+interface NexusPrismaOutputs {
+  Query: {
+    modelName: 'ModelName'
+    modelNames: 'ModelName'
+  },
+  Mutation: {
+    createOneModelName: 'ModelName'
+    updateOneModelName: 'ModelName'
+    updateManyModelName: 'BatchPayload'
+    deleteOneModelName: 'ModelName'
+    deleteManyModelName: 'BatchPayload'
+    upsertOneModelName: 'ModelName'
+  },
+  ModelName: {
+    id: 'Int'
+    name: 'String'
+    amount: 'Int'
+  }
+}
+
+// Helper to gather all methods relative to a model
+interface NexusPrismaMethods {
+  ModelName: Typegen.NexusPrismaFields<'ModelName'>
+  Query: Typegen.NexusPrismaFields<'Query'>
+  Mutation: Typegen.NexusPrismaFields<'Mutation'>
+}
+
+interface NexusPrismaGenTypes {
+  inputs: NexusPrismaInputs
+  outputs: NexusPrismaOutputs
+  methods: NexusPrismaMethods
+  models: PrismaModels
+  pagination: Pagination
+  scalars: CustomScalars
+}
+
+declare global {
+  interface NexusPrismaGen extends NexusPrismaGenTypes {}
+
+  type NexusPrisma<
+    TypeName extends string,
+    ModelOrCrud extends 'model' | 'crud'
+  > = Typegen.GetNexusPrisma<TypeName, ModelOrCrud>;
+}
+  "
+`;
+
+exports[`support atomic operations (default): schema 1`] = `
+"type Mutation {
+  createOneModelName(data: ModelNameCreateInput!): ModelName!
+  deleteOneModelName(where: ModelNameWhereUniqueInput!): ModelName
+  updateOneModelName(data: ModelNameUpdateInput!, where: ModelNameWhereUniqueInput!): ModelName
+  upsertOneModelName(where: ModelNameWhereUniqueInput!, create: ModelNameCreateInput!, update: ModelNameUpdateInput!): ModelName!
+  updateManyModelName(data: ModelNameUpdateManyMutationInput!, where: ModelNameWhereInput): BatchPayload!
+}
+
+type ModelName {
+  id: Int!
+}
+
+input ModelNameCreateInput {
+  name: String!
+  amount: Int!
+}
+
+input ModelNameWhereUniqueInput {
+  id: Int
+}
+
+input ModelNameUpdateInput {
+  name: StringFieldUpdateOperationsInput
+  amount: IntFieldUpdateOperationsInput
+}
+
+type BatchPayload {
+  count: Int!
+}
+
+input ModelNameUpdateManyMutationInput {
+  name: StringFieldUpdateOperationsInput
+  amount: IntFieldUpdateOperationsInput
+}
+
+input ModelNameWhereInput {
+  AND: [ModelNameWhereInput!]
+  OR: [ModelNameWhereInput!]
+  NOT: [ModelNameWhereInput!]
+  id: IntFilter
+  name: StringFilter
+  amount: IntFilter
+}
+
+input StringFieldUpdateOperationsInput {
```

Comment:
Actually a problem here. Disabling atomic operations should remove the input object nesting.

Was this approach intentional on your part?

Reply:
Hm, considering what you wrote in the docs here https://github.com/graphql-nexus/schema/pull/587/files#diff-aca15a6355a7aef9e2bd3976c18aa15715429c456cea7531795eabc1d0b572b3 I believe these tests results were not your intention.

Reply:
There is two tests in this file: default and disabled. You comment the result when atomic operations are enabled (by default). I can remove this default test and keep only the disabled if this is not clear